src/main.rs

```diff
@@ -28,10 +28,25 @@ fn main() -> ExitCode {
         to,
         context_count,
         format,
+        report_identical_files,
     } = parse_params(opts).unwrap_or_else(|error| {
         eprintln!("{error}");
         exit(2);
     });
+    // if from and to are the same file, no need to perform any comparison
+    let maybe_report_identical_files = || {
```

Comment:
maybe a dumb question but why not moving this into a function?

Reply:
I used [a closure](https://doc.rust-lang.org/book/ch13-01-closures.html) because unlike nested functions, closures can capture values from the scope in which they’re defined.

With a nested function, I would have had to pass `report_identical_files`, `from` and `to` as parameters.

If you prefer a function though, I'm fine with changing this, I don't have a strong opinion either way.

Reply:
i was curious